ruff.toml

```diff
@@ -0,0 +1,29 @@
+preview = true
+
+extend-exclude = []
```

Comment:
Do we use any feature that is in preview? If not this can be gone

Reply:
Yes, we do. I noted in the ruff.toml which ones require the preview flag.